libzmix/bulletproofs_amcl/LICENSE

```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```

Comment:
@lovesh can you please clarify the origin of this code?
constraint_system.rs for example appears similar to https://gitlab.dusk.network/kev/bp/blob/7ca35bb2a0ff2d1e7de7aaea7b3faf15ff2856e4/src/r1cs/constraint_system.rs
That code appears under an MIT license. 
If this is the case please contact @dhuseby 
He can explain whether and how to use third party code and licensing obligations.

Reply:
What is meant by origin? Like copy pasted the code as is or following a certain design. The design is copied from (there are some differences though) from dalek's [Bulletproofs implementation here](https://github.com/dalek-cryptography/bulletproofs/blob/develop/src/r1cs/constraint_system.rs). That implementation is MIT.

Reply:
So to clarify, this is a fork of the dalek's bp implementation? Do you have any intention to upstream our patches to dalek's code? My recommendation is that we should upstream and then just have a submodule reference to dalek's repo. That adds a bunch of overhead that is a headache but it clarifies a lot of things. IIRC, MIT licensed code is considered "compatible" with HL licensing policies. I need to double check on whether we need a license exception on that code. I don't think so but I want to be certain.

Reply:
This is not a fork but reimplementation (not completely as it is though) of their design using a different math library. Our code is thus incompatible with theirs and we cannot use dalek as a dependency.